client/renderSDL/SDLImage.cpp

```diff
@@ -274,9 +277,15 @@ std::shared_ptr<ISharedImage> SDLImageShared::scaleInteger(int factor, SDL_Palet
 	if (palette && surf && surf->format->palette)
 		SDL_SetSurfacePalette(surf, palette);
 
-	SDL_Surface * scaled = CSDL_Ext::scaleSurfaceIntegerFactor(surf, factor, EScalingAlgorithm::XBRZ);
+	SDL_Surface * scaled = nullptr;
+	if(preScaleFactor == factor)
+		scaled = CSDL_Ext::scaleSurfaceIntegerFactor(surf, 1, EScalingAlgorithm::NEAREST); // keep size
```

Comment:
I think in this case we can safely return this (or well, shared_from_this) and skip copy creation

Reply:
`error: could not convert ‘std::enable_shared_from_this<_Tp>::shared_from_this() const [with _Tp = SDLImageShared]()’ from ‘shared_ptr<const SDLImageShared>’ to ‘shared_ptr<ISharedImage>’`

Reply:
Ah, this is const method that returns non-const value, so can't return non-const this.
A bit of a hack, but should work:
```suggestion
		scaled = std::const_pointer_cast<const SDLImageShared>(shared_from_this());
```

Although we should probably replaces all `std::shared_ptr<ISharedImage>` with `std::shared_ptr<const ISharedImage>`

Reply:
Also not possible...

It's also not possible to re-use surf pointer...

Replaced it with simple copy...